numpy/_core/fromnumeric.py

```diff
@@ -655,6 +655,41 @@ def transpose(a, axes=None):
     return _wrapfunc(a, 'transpose', axes)
 
 
+def _matrix_transpose_dispatcher(x):
+    return (x,)
+
+@array_function_dispatch(_matrix_transpose_dispatcher)
+def matrix_transpose(x, /):
+    """
+    Transposes a matrix (or a stack of matrices) ``x``.
+
+    This function is Array API compatible.
+
+    Parameters
+    ----------
+    x : array_like
+        Input array having shape (..., M, N) and whose two innermost
+        dimensions form ``MxN`` matrices.
+
+    Returns
+    -------
+    out : ndarray
+        An array containing the transpose for each matrix and having shape
+        (..., N, M).
+
+    See Also
+    --------
+    transpose : Generic transpose method.
+
+    """
+    x = asarray(x)
```

Comment:
Similar to the discussion in the other PR - should thise be `asanyarray`?

Reply:
I think there was roughly an agreement in https://github.com/numpy/numpy/pull/25101#discussion_r1419879122 that `asanyarray` can be used in place of `asarray`.

I replaced it for functions added here and `outer` that has been merged.